src/proc/color-formats-converter.cpp

```diff
@@ -46,6 +46,294 @@ bool has_avx()
+            auto reg_num_on_line = w * rgb_bpp / 16;
+            auto line = (i*num_on_regs_at_once) / reg_num_on_line;
+            auto j = i % (reg_num_on_line / num_on_regs_at_once);
+
+            // Align registers and store 16 pixels (48 bytes) at once on the line above
+            _mm_storeu_si128(&dst[(line*2 ) *reg_num_on_line + j * 3], _mm_alignr_epi8(rgb1_l0, rgb0_l0, 4));
+            _mm_storeu_si128(&dst[(line*2 ) * reg_num_on_line + j * 3 + 1], _mm_alignr_epi8(rgb2_l0, rgb1_l0, 8));
+            _mm_storeu_si128(&dst[(line*2 ) * reg_num_on_line + j * 3 + 2], _mm_alignr_epi8(rgb3_l0, rgb2_l0, 12));
+
+            // Shuffle rgb triples to the start and end of each register
+            __m128i rgb0_l1 = _mm_shuffle_epi8(rgba_0_7_l1, _mm_setr_epi8(3, 7, 11, 15, 0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14));
+            __m128i rgb1_l1 = _mm_shuffle_epi8(rgba_8_15_l1, _mm_setr_epi8(0, 1, 2, 4, 3, 7, 11, 15, 5, 6, 8, 9, 10, 12, 13, 14));
+            __m128i rgb2_l1 = _mm_shuffle_epi8(rgba_16_23_l1, _mm_setr_epi8(0, 1, 2, 4, 5, 6, 8, 9, 3, 7, 11, 15, 10, 12, 13, 14));
+            __m128i rgb3_l1 = _mm_shuffle_epi8(rgba_24_32_l1, _mm_setr_epi8(0, 1, 2, 4, 5, 6, 8, 9, 10, 12, 13, 14, 3, 7, 11, 15));
+
+            // Align registers and store 16 pixels(48 bytes) at once on the line bellow
+            _mm_storeu_si128(&dst[(line*2  + 1) *reg_num_on_line + j * 3], _mm_alignr_epi8(rgb1_l1, rgb0_l1, 4));
+            _mm_storeu_si128(&dst[(line*2  + 1) * reg_num_on_line + j * 3 + 1], _mm_alignr_epi8(rgb2_l1, rgb1_l1, 8));
+            _mm_storeu_si128(&dst[(line*2  + 1) * reg_num_on_line + j * 3 + 2], _mm_alignr_epi8(rgb3_l1, rgb2_l1, 12));
+        }
+#endif
+    }
+    void unpack_y411_native(byte * const dest, const byte * s, int w, int h, int actual_size)
+    {
+        auto out = dest;
+
+        auto index_source = 0;
+        for (auto i = 0; i < h; i += 2)
+        {
+            for (auto j = 0; j < w; j += 2)
+            {
+                auto y411_pix = &s[index_source];
+                auto l0_u0 = y411_pix[0];
+                auto l0_y0 = y411_pix[1];
+                auto l0_y1 = y411_pix[2];
+                auto l0_v0 = y411_pix[3];
+                auto l1_y0 = y411_pix[4];
+                auto l1_y1 = y411_pix[5];
+
+                byte yuv0_0[3] = { l0_y0, l0_u0, l0_v0 };
+                convert_yuv_to_rgb(yuv0_0, &out[i * w * 3 + j * 3]);
+
+                byte yuv0_1[3] = { l0_y1, l0_u0, l0_v0 };
+                convert_yuv_to_rgb(yuv0_1, &out[i * w * 3 + j * 3 + 3]);
+
+                byte yuv1_0[3] = { l1_y0, l0_u0, l0_v0 };
+                convert_yuv_to_rgb(yuv1_0, &out[(i + 1) * w * 3 + j * 3]);
+
+                byte yuv1_1[3] = { l1_y1, l0_u0, l0_v0 };
+                convert_yuv_to_rgb(yuv1_1, &out[(i + 1) * w * 3 + j * 3 + 3]);
+
+                index_source += 6;
+            }
+        }
+    }
+    void unpack_y411( byte * const dest[], const byte * s, int w, int h, int actual_size )
```

Comment:
Explain... what does unpacking do?

Reply:
done

Reply:
There are constraints, too, right? These should be specified...

Reply:
constraints?

Reply:
done